specs/beacon-chain.md

```diff
@@ -870,6 +905,81 @@ while len(crystallized_state.persistent_committee_reassignments) > 0 and crystal
     crystallized_state.persistent_committees[rec.shard].append(rec.validator_index)
 ```
 
+## Shard block processing
+
+A block on a shard can be processed only if its `parent` has been accepted. To validate a block header on shard `shard_id`, compute as follows:
+
+* Verify that `beacon_chain_ref` is the hash of the `slot`'th block in the beacon chain.
+* Let `state` be the state of the beacon chain block referred to by `beacon_chain_ref`. Let `validators` be `[validators[i] for i in state.current_persistent_committees[shard_id]]`.
+* Assert `len(attester_bitfield) == ceil_div8(len(validators))`
+* Let `curblock_proposer_index = hash(state.randao_mix + bytes8(shard_id)) % len(validators)`. Let `parent_proposer_index` be the same value calculated for the parent block.
+* Make sure that the `parent_proposer_index`'th bit in the `attester_bitfield` is set to 1.
+* Generate the group public key by adding the public keys of all the validators for whom the corresponding position in the bitfield is set to 1. Verify the `aggregate_sig` using this as the pubkey and the `parent_hash` as the message.
```

Comment:
So the only opportunity for shard block signatures to be included is in the immediate following block?

Reply:
but fork choice is still based upon signatures seen on the wire rather than just those included in the block?